kodiak/test_evaluation.py

```diff
@@ -3334,3 +3334,257 @@ def test_get_merge_body_empty(pull_request: PullRequest) -> None:
+    assert api.set_status.call_count == 0
+    assert api.dequeue.call_count == 0
+
+
+@pytest.mark.asyncio
+async def test_mergeable_passing_update_always_enabled(
+    api: MockPrApi,
+    config: V1,
+    config_path: str,
+    config_str: str,
+    pull_request: PullRequest,
+    branch_protection: BranchProtectionRule,
+    review: PRReview,
+    context: StatusContext,
+    check_run: CheckRun,
+) -> None:
+    """
+    Test happy case with update.always enabled. We should shouldn't see any
+    difference with update.always enabled.
+    """
+    config.update.always = True
+    api.queue_for_merge.return_value = 3
+    await mergeable(
+        api=api,
+        config=config,
+        config_str=config_str,
+        config_path=config_path,
+        pull_request=pull_request,
+        branch_protection=branch_protection,
+        review_requests=[],
+        reviews=[review],
+        contexts=[context],
+        check_runs=[check_run],
+        valid_signature=False,
+        valid_merge_methods=[MergeMethod.squash],
+        merging=False,
+        is_active_merge=False,
+        skippable_check_timeout=5,
+        api_call_retry_timeout=5,
+        api_call_retry_method_name=None,
+    )
+    assert api.set_status.call_count == 1
+    assert "enqueued for merge (position=4th)" in api.set_status.calls[0]["msg"]
+    assert api.queue_for_merge.call_count == 1
+    assert api.dequeue.call_count == 0
+
+
+@pytest.mark.asyncio
+async def test_mergeable_update_always_enabled_merging_behind_pull_request(
+    api: MockPrApi,
+    config: V1,
+    config_path: str,
+    config_str: str,
+    pull_request: PullRequest,
+    branch_protection: BranchProtectionRule,
+    review: PRReview,
+    context: StatusContext,
+    check_run: CheckRun,
+) -> None:
+    """
+    When we're merging with update.always enabled we don't want to update the
+    branch using our update.always logic. We want to update using our merging
+    logic so we trigger the PollForever exception necessary to continue our
+    merge loop. If we used the update.always logic we'd eject a PR if it became
+    out of sync during merge.
+    """
+    config.update.always = True
+    pull_request.mergeStateStatus = MergeStateStatus.BEHIND
+    branch_protection.requiresStrictStatusChecks = True
+
+    with pytest.raises(PollForever):
+        await mergeable(
+            api=api,
+            config=config,
+            config_str=config_str,
+            config_path=config_path,
+            pull_request=pull_request,
+            branch_protection=branch_protection,
+            review_requests=[],
+            reviews=[review],
+            contexts=[context],
+            check_runs=[check_run],
+            valid_signature=False,
+            valid_merge_methods=[MergeMethod.squash],
+            is_active_merge=False,
+            skippable_check_timeout=5,
+            api_call_retry_timeout=5,
+            api_call_retry_method_name=None,
+            #
```

Comment:
nit

```suggestion
```

Reply:
I'm using those `#`s to mark the arguments that are customized.